nuxeo-common/src/main/java/org/nuxeo/common/utils/PeriodAndDuration.java

```diff
@@ -0,0 +1,220 @@
+                    int months = parseInt(monthMatch, text, "months");
+                    int days = parseInt(dayMatch, text, "days");
+                    long hoursAsSecs = parseNumber(hourMatch, SECONDS_PER_HOUR, text, "hours");
+                    long minsAsSecs = parseNumber(minuteMatch, SECONDS_PER_MINUTE, text, "minutes");
+                    long seconds = parseNumber(secondMatch, 1, text, "seconds");
+                    int nanos = parseFraction(fractionMatch, Long.signum(seconds), text);
+                    try {
+                        Period period = Period.of(years, months, days);
+                        if (negate) {
+                            period = period.negated();
+                        }
+                        seconds = Math.addExact(hoursAsSecs, Math.addExact(minsAsSecs, seconds));
+                        Duration duration = Duration.ofSeconds(seconds, nanos);
+                        if (negate) {
+                            duration = duration.negated();
+                        }
+                        return new PeriodAndDuration(period, duration);
+                    } catch (ArithmeticException ex) {
+                        throw (DateTimeParseException) new DateTimeParseException(
+                                "Text cannot be parsed to a PeriodAndDuration: overflow", text, 0).initCause(ex);
+                    }
+                }
+            }
+        }
+        throw new DateTimeParseException("Text cannot be parsed to a PeriodAndDuration", text, 0);
+    }
+
+    protected static int parseInt(String string, String text, String errorText) {
+        if (string == null) {
+            return 0;
+        }
+        try {
+            return Integer.parseInt(string);
+        } catch (NumberFormatException e) {
+            throw (DateTimeParseException) new DateTimeParseException(
+                    "Text cannot be parsed to a PeriodAndDuration: " + errorText, text, 0).initCause(e);
+        }
+    }
+
+    protected static long parseNumber(String string, int multiplier, String text, String errorText) {
+        if (string == null) {
+            return 0;
+        }
+        try {
+            long val = Long.parseLong(string);
+            return Math.multiplyExact(val, multiplier);
+        } catch (NumberFormatException | ArithmeticException e) {
+            throw (DateTimeParseException) new DateTimeParseException(
+                    "Text cannot be parsed to a PeriodAndDuration: " + errorText, text, 0).initCause(e);
+        }
+    }
+
+    protected static int parseFraction(String string, int sign, String text) {
+        if (string == null || string.length() == 0) {
+            return 0;
+        }
+        try {
+            string = (string + "000000000").substring(0, 9);
+            return Integer.parseInt(string) * sign;
+        } catch (NumberFormatException | ArithmeticException e) {
+            throw (DateTimeParseException) new DateTimeParseException(
```

Comment:
I'm probably missing something, why not using the constructor taking a Throwable?

Reply:
Indeed, I was just using the same code as `java.time.Duration`. I'll fix.